Bottom: 41275ebf36f106cedef6963331ebce848338db3a
Top:    c271dda29a135a715e69d0061f0e7a93c9215f4c
Author: Gabriel Paubert <paubert@iram.es>
Date:   2011-05-17 11:44:01 +0100

Lesstif + GTK HIDs: Ignore ISO Level3 shift modifier key

This key press is found on some international keyboards (e.g. Spanish).
If we don't ignore it, PCB complains about the key not being tied to
an action when it is pressed.

From the committer, Peter Clifton <pcjc2@cam.ac.uk>
...
Thanks to Gabriel for reporting this and providing the patch for
Lesstif. I have updated the patch to make the equivalent change to
the GTK HID as well.

Signed-off-By: Peter Clifton <pcjc2@cam.ac.uk>


---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index d969dd2..5f5f16e 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -345,6 +345,7 @@ ghid_port_key_press_cb (GtkWidget * drawing_area,
     case GDK_Shift_L:
     case GDK_Shift_R:
     case GDK_Shift_Lock:
+    case GDK_ISO_Level3_Shift:
       break;
 
     case GDK_Up:
diff --git a/src/hid/lesstif/menu.c b/src/hid/lesstif/menu.c
index 297768b..8de2952 100644
--- a/src/hid/lesstif/menu.c
+++ b/src/hid/lesstif/menu.c
@@ -1140,6 +1140,7 @@ lesstif_key_event (XKeyEvent * e)
     case XK_Super_R:
     case XK_Hyper_L:
     case XK_Hyper_R:
+    case XK_ISO_Level3_Shift:
       return 1;
     }
